### PR TITLE
Introduce `findEntry_` and avoid examining `MutVar` twice in `at`

### PR DESCRIPTION
**Pull request by Bodigrim**
At the moment `at` unwraps `MutVar` twice:
https://github.com/klapaucius/vector-hashtables/blob/2dc4c0dc2d9471bec8f1d1b8edd161fcc78250fc/src/Data/Vector/Hashtables/Internal.hs#L234-L242

`findEntry` runs `readMutVar . getDRef` under the hood, but once an entry is found `at` has to do `readMutVar . getDRef` again. While I don't think this is a significant bottleneck, it would be better to unwrap `MutVar` only once, which is achieved in this PR by introducing `findEntry_` helper. 

The latter serves also as a solution (at least a partial one) to #25: `findEntry_` operates on underlying `Dictionary_` and provides the most important part of the puzzle for other operations, so if a low-level client is desparate enough to squeeze every last bit of performance, they can.

I also made a couple of minor documentation fixes and changes.

----

If this looks good, I'd appreciate a release soon.

### CODE REVIEW

**Comment by Bodigrim** [comment]
@swamp-agr @klapaucius @ulysses4ever just a gentle reminder.

**Comment by ulysses4ever** [comment]
It's on my radar but I'm having two weeks of deadlines after deadlines, so can't get to it before next week perhaps. Sorry!

**Review by ulysses4ever — APPROVED** [review]
Terrific, thank you!

File: src/Data/Vector/Hashtables.hs [hunk 1]
```diff
@@ -16,7 +16,7 @@ module Data.Vector.Hashtables
     , FrozenDictionary (..)
     , findElem
     , Dictionary_ (..)
-    , findEntry
+    , findEntry_
```
[the remark this answers is not in the document]
Comment (Bodigrim) [reply src/Data/Vector/Hashtables.hs:19]: "Low-level interface" makes sense, but in such case we should probably move the entire section to the bottom?..

[the remark this answers is not in the document]
Comment (ulysses4ever) [reply src/Data/Vector/Hashtables.hs:19]: I'd agree.

[the remark this answers is not in the document]
Comment (Bodigrim) [reply src/Data/Vector/Hashtables.hs]: Let's do it, hopefully it would not cause too much bikeshedding ;)

**Comment by ulysses4ever** [comment]
Quick update. There’s some internal talk going on about whether this extra layer of functions may not inline as well. Needs benchmarking. Or duplicating the code manually instead of calling findEntry_.

**Comment by Bodigrim** [comment]
@ulysses4ever you can add 

```haskell
{-# OPTIONS_GHC -ddump-to-file -ddump-simpl -dsuppress-all -dno-suppress-type-signatures #-}
```

to dump Core and compare. Namely, `findEntry` / `at` / `at'` remain exactly the same as it used to be in `master` modulo variable names.

**Comment by Bodigrim** [comment]
Thanks! It would be great to have a new release of `vector-hashtables` now :)

**Comment by ulysses4ever** [comment]
@Bodigrim great idea, and indeed it is planned! Hopefully, this week.

**Comment by swamp-agr** [comment]
@Bodigrim many thanks for your awesome contributions! Please expect release soon.

**Pull request merged**